ksqldb-engine/src/main/java/io/confluent/ksql/connect/ConnectRequestHeadersExtension.java

```diff
@@ -25,16 +27,44 @@
  */
 public interface ConnectRequestHeadersExtension {
 
+  /**
+   * Returns whether to use the custom auth header returned from
+   * {@link ConnectRequestHeadersExtension#getAuthHeader(List)} instead of
+   * ksqlDB's default auth header for ksqlDB connector requests. This auth header
+   * configured by this extension takes precedence over any other custom auth
+   * configurations for ksqlDB connector requests, including basic auth configured in
+   * ksqlDB server properties.
+   *
+   * @return whether to use the custom auth header returned from
+   *         {@link ConnectRequestHeadersExtension#getAuthHeader(List)} instead of
+   *         ksqlDB's default auth header for ksqlDB connector requests
+   */
+  default boolean shouldUseCustomAuthHeader() {
```

Comment:
In what situations `ConnectRequestHeadersExtension` is configured, but it should not be used?

Reply:
If someone wants to only use the extension for custom additional headers (via the `getHeaders()` method) but wants to continue using ksqlDB's default auth header, then they can set `shouldUseCustomAuthHeader = false` rather than re-implementing ksqlDB's logic for computing the default auth header.

Reply:
I can add a line to the javadoc to clarify.

Reply:
Thank you! This makes sense.